### PR TITLE
[16.0][FIX] mrp_subcontracting_skip_no_negative: consider move with multiple productions

### PR DESCRIPTION
**Pull request by JordiMForgeFlow**

### CODE REVIEW

File: mrp_subcontracting_skip_no_negative/models/stock_move.py [hunk 1]
```diff
@@ -16,9 +16,9 @@ def _action_done(self, cancel_backorder=False):
         for move in moves_with_no_check:
             production_move = self.search([("move_dest_ids", "=", move.id)])
             production = production_move.production_id
-            if production.reservation_state != "assigned":
+            if production and production[0].reservation_state != "assigned":
```
Comment (yostashiro) [inline mrp_subcontracting_skip_no_negative/models/stock_move.py]: Wouldn't it be more appropriate to do something like this?
```suggestion
            if any(state != "assigned" for state in productions.mapped("reservation_state")):
```

Comment (JordiMForgeFlow) [reply mrp_subcontracting_skip_no_negative/models/stock_move.py]: @yostashiro Could you check now? What I have done is to look for pending production moves. If there are several productions, I first try to assign the ones unassigned, and then check that they are all assigned

**Review by DavidJForgeFlow — APPROVED** [review]
LGTM :+1: Code review

**Review by AaronHForgeFlow — APPROVED** [review]
code LGTM

**Review by MiquelRForgeFlow — APPROVED** [review]

File: mrp_subcontracting_skip_no_negative/models/stock_move.py [hunk 2]
```diff
@@ -14,11 +14,21 @@ def _action_done(self, cancel_backorder=False):
...
+                [
+                    ("move_dest_ids", "=", move.id),
+                    ("state", "not in", ("done", "cancel")),
+                ]
+            )
+            productions = production_move.production_id
+            unassigned_productions = productions.filtered(
+                lambda p: p.reservation_state != "assigned"
+            )
+            for production in unassigned_productions:
```
Comment (yostashiro) [inline mrp_subcontracting_skip_no_negative/models/stock_move.py]:
```suggestion
            unassigned_productions.action_assign()
```
and remove the following line.

File: mrp_subcontracting_skip_no_negative/models/stock_move.py [hunk 3]
```diff
@@ -14,11 +14,21 @@ def _action_done(self, cancel_backorder=False):
...
                 production.action_assign()
-            if production.reservation_state == "assigned":
+            if all(
+                state == "assigned" for state in productions.mapped("reservation_state")
```
Comment (yostashiro) [inline mrp_subcontracting_skip_no_negative/models/stock_move.py]: I guess this is a trivial point and may not matter.
```suggestion
                state == "assigned" for state in unassigned_productions.mapped("reservation_state")
```

**Review by yostashiro — APPROVED** [review]
Thanks for the update!

File: mrp_subcontracting_skip_no_negative/models/stock_move.py (hunk 3 above)
Comment (JordiMForgeFlow) [reply mrp_subcontracting_skip_no_negative/models/stock_move.py]: it looks much better like this, thanks!

**Review by AungKoKoLin1997 — APPROVED** [review]
Code Review: LGTM

**Review by BernatPForgeFlow — APPROVED** [review]
LGTM!

**Pull request merged**